src/bokeh/util/sampledata.py

```diff
@@ -56,39 +56,48 @@ def download(progress: bool = True) -> None:
 
     # HTTP requests are cheaper for us, and there is nothing private to protect
     s3 = 'http://sampledata.bokeh.org'
-    files = json.load(open(Path(__file__).parent / "sampledata.json"))
+    files = metadata()
 
-    for filename, md5 in files:
-        real_name, ext = splitext(filename)
-        if ext == '.zip':
-            if not splitext(real_name)[1]:
-                real_name += ".csv"
-        else:
-            real_name += ext
-        real_path = data_dir /real_name
+    for file_name, md5 in files.items():
+        real_path = data_dir / real_name(file_name)
 
         if real_path.exists():
-            local_md5 = hashlib.md5(open(real_path, 'rb').read()).hexdigest()
+            with open(real_path, "rb") as file:
+                data = file.read()
+            local_md5 = hashlib.md5(data).hexdigest()
             if local_md5 == md5:
-                print(f"Skipping {filename!r} (checksum match)")
+                print(f"Skipping {file_name!r} (checksum match)")
                 continue
             else:
-                print(f"Re-fetching {filename!r} (checksum mismatch)")
+                print(f"Re-fetching {file_name!r} (checksum mismatch)")
 
-        _download_file(s3, filename, data_dir, progress=progress)
+        _download_file(s3, file_name, data_dir, progress=progress)
```

Comment:
I'm not sure if it is actually important to convey that this is a re-fetch. I'd support a simplification
```python
            if local_md5 == md5:
                print(f"Skipping {file_name!r} (checksum match)")
                continue

         print(f"Fetching {file_name!r}")
        _download_file(s3, file_name, data_dir, progress=progress)
```